hnn_core/network.py

```diff
@@ -14,6 +15,52 @@
 from .params import create_pext
 
 
+def read_spikes(fname, gid_dict=None):
+    """Read spiking activity from a collection of spike trial files.
+
+    Parameters
+    ----------
+    fname : str
+        Wildcard expression (e.g., '<pathname>/spk_*.txt') of the
+        path to the spike file(s).
+    gid_dict : dict of lists or range objects | None
+        Dictionary with keys 'evprox1', 'evdist1' etc.
+        containing the range of Cell or input IDs of different
+        cell or input types. If None, each spike file must contain
+        a 3rd column for spike type.
+
+    Returns
+    ----------
+    spikes : Spikes
+        An instance of the Spikes object.
+    """
+
+    spike_times = []
+    spike_gids = []
+    spike_types = []
+    for file in sorted(glob(fname)):
+        spike_trial = np.loadtxt(file, dtype=str)
+        spike_times += [list(spike_trial[:, 0].astype(float))]
+        spike_gids += [list(spike_trial[:, 1].astype(int))]
+
+        # Note that legacy HNN 'spk.txt' files don't contain a 3rd column for
+        # spike type. If reading a legacy version, validate that a gid_dict is
+        # provided.
+        if spike_trial.shape[1] == 3:
+            spike_types += [list(spike_trial[:, 2].astype(str))]
+        else:
+            if gid_dict is None:
```

Comment:
```suggestion
        elif gid_dict is None:
```

sorry I didn't notice this before. Avoid nesting when possible. See:

```sh
$ python -c 'import this' | head -7 | tail -1
Flat is better than nested
```

Reply:
Oh good point, idk how I missed that. Any other stylistic changes I can make in this batch?